source/Nevermore/Util/DataModificationQueryBuilder.cs

```diff
@@ -420,34 +448,20 @@ CommandParameterValues GetDocumentParameters(Func<DocumentMap, object> allocateI
             sb.AppendLine(statement);
             sb.AppendLine();
 
-            if (relatedDocumentData.Any(d => d.Related.Any()))
-                sb.AppendLine("DECLARE @references as TABLE (Reference nvarchar(400), ReferenceTable nvarchar(400))");
+            if (!ShouldUseTableValuedParameterForUpdate() && relatedDocumentData.Any(d => d.Related.Any()))
+                sb.AppendLine($"DECLARE @references as TABLE ([{RelatedDocumentTableValuedParameterRelatedDocumentIdColumn}] nvarchar(400), [{RelatedDocumentTableValuedParameterRelatedDocumentTableColumn}] nvarchar(400))");
```

Comment:
Renamed the columns in this temp table to be the same as our table valued parameter table type, so we can reuse the query